pkg/sbproxy/sbproxy.go

```diff
@@ -0,0 +1,227 @@
+package sbproxy
+
+import (
+	"context"
+	"strconv"
+	"sync"
+
+	"os"
+	"os/signal"
+
+	"time"
+
+	"fmt"
+
+	"net/http"
+
+	"github.com/Peripli/service-broker-proxy/pkg/osb"
+	"github.com/Peripli/service-broker-proxy/pkg/platform"
+	"github.com/Peripli/service-broker-proxy/pkg/sbproxy/middleware"
+	"github.com/Peripli/service-broker-proxy/pkg/sm"
+	"github.com/Peripli/service-broker-proxy/pkg/task"
+	"github.com/gorilla/mux"
+	"github.com/pmorie/osb-broker-lib/pkg/metrics"
+	"github.com/pmorie/osb-broker-lib/pkg/rest"
+	"github.com/pmorie/osb-broker-lib/pkg/server"
+	prom "github.com/prometheus/client_golang/prometheus"
+	"github.com/robfig/cron"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	BrokerPathParam = "brokerID"
+	ApiPrefix       = "/v1/osb/{" + BrokerPathParam + "}"
+)
+
+var (
+	ctx    context.Context
+	cancel context.CancelFunc
+	group  sync.WaitGroup
+)
```

Comment:
why these global vars? why not move them inside `SBProxy`?

Reply:
They are package private - the reason is that i thought the code looks cleaner as i dont have to pass them around as parameters of the methods in the file - i can move them in - let me know what you think wouldbe cleaner